File: back/back/Models/Systeme.cs
using LiteDB;

namespace back.Models;

public sealed class Systeme
{
    [BsonId]
    public int Id { get; set; }
    public string Nom { get; set; } = null!;
    public string? Description { get; set; }

    public decimal PositionX { get; set; }
    public decimal PositionY { get; set; }
}

File: front/src/models/Systeme.ts
export type Systeme =
{
    id: number,
    nom: string,
    description: string | null,

    positionX: number,
    positionY: number
}

export type SystemeConnecter = 
{
    idSystemeA: number,
    idSystemeB: number,
    distance: string | null
}

export type SystemeLeger =
{
    id: number,
    nom: string
}

export type SystemePositionRequete =
{
    positionX: number,
    positionY: number
}

export type SystemeRequete = Omit<Systeme, "id">;
export type SystemeConnecterSupprimerRequete = Omit<SystemeConnecter, "distance">;